### PR TITLE
Switched to SLF4J for logging

### PR DESCRIPTION
**Pull request by MinnDevelopment**
This change will bump the version to 3.3.1

### CODE REVIEW

**Comment by Shredder121** [comment]
Is it possible to add some more methods that accept format arguments?
So when TRACE is not enabled, there is no string concatenation done?
(When all events come in, the message is concatenated, but not actually printed, it will save resources and be faster to only do it on demand)

Also, (Simple)DateFormat isn't thread safe.
(I know it might not be possible to change it in this PR, but I'd recommend fixing that)

**Comment by MinnDevelopment** [comment]
This is the first iteration of completely replacing logging.
In this step we are adding first support for SLF4J and in the second step we will be removing SimpleLog completely in favor of the Logger interface. In that step we can make use of the SLF4J formatting methods.

**Comment by Shredder121** [comment]
👍 Just wanted to make sure, carry on.

**Comment by MinnDevelopment** [comment]
This will be merged once categories have been released on the stable discord client to be in sync with the stable category release when bumping the version.

File: src/main/java/net/dv8tion/jda/core/utils/SimpleLog.java [hunk 1]
```diff
@@ -290,53 +105,113 @@ public Level getLevel()
...
+        {
+            if (logger.isTraceEnabled())
+                return Level.TRACE;
+            if (logger.isDebugEnabled())
+                return Level.DEBUG;
+            if (logger.isInfoEnabled())
+                return Level.INFO;
+            if (logger.isWarnEnabled())
+                return Level.WARN;
+            return Level.ERROR;
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/utils/SimpleLog.java]: is there no OFF level?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/core/utils/SimpleLog.java]: No there is not

**Pull request merged**